cli/dcoscli/package/main.py

```diff
@@ -499,16 +499,20 @@ def _uninstall(package_name, remove_all, app_id, cli, app, skip_confirmation):
     :rtype: int
     """
 
+    package_manager = get_package_manager()
+
     # Don't gate CLI uninstalls.
     # Don't gate uninstalls if the user wants to skip confirmation.
-    if not cli and not skip_confirmation and not _confirm_uninstall(
-        package_name,
-        remove_all,
-        app_id
-    ):
-        return 0
+    if not cli and not skip_confirmation:
+        installed = package.installed_packages(
+            package_manager, None, package_name, cli_only=False)
+        installed_pkg = next(iter(installed), None)
+
+        # Don't gate implicit CLI uninstalls.
```

Comment:
I think we should still gate them, but with a different prompt. Basically informing them that this CLI will be removed for all instances of interacting with an App (i.e. in the case of installing say the Kafka CLI) and forcing them to confirm with a Yes/No.

Reply:
As an experienced cli user, I don't want a yes/no confirm behavior for uninstalling cli packages, but I may not represent most users.

I think for `dcos package uninstall foo --cli` you would never want this behavior.  However, for `dcos package uninstall foo` I'm less sure (which I guess is this case.)  It's also hard to describe this accurately: "you'll lose this cli package for all services it might relate to, even though it might not be a service-related extension, and also it will only be lost for this cluster and not other attached clusters."

------

I also hope someone smarter than me knows this boolean construction can't engage for `dcos package uninstall foo` that targets cli and an app both.

Reply:
I'll use a different prompt message, to bypass it there still would be the `--yes` option.